src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs

```diff
@@ -3045,6 +3045,142 @@ public static void XmlMembersMapping_CompositeType()
         Assert.Equal(requestBodyValue.composite.StringValue, requestBodyActual.composite.StringValue);
     }
 
+    [Fact]
+    public static void Xml_HiddenDerivedFieldTest()
+    {
+        using (var ms = new MemoryStream())
+        {
+            var derived = new DerivedClass { value = "on derived" };
+            var ser = new XmlSerializer(typeof(BaseClass));
+            ser.Serialize(ms, derived);
+
+            ms.Position = 0;
+            var roundtripped = ser.Deserialize(ms) as BaseClass;
+
+            if (roundtripped.value == null && roundtripped.Value == null &&
+                ((DerivedClass)roundtripped).Value == null &&
+                ((DerivedClass)roundtripped).value == derived.value)
+            {
+                Assert.True(true);
+            }
+            else
+            {
+                Assert.True(false, "Roundtrip verification for HiddenDerivedField test failed");
+            }
+        }
+    }
+
+    [Fact]
+    public static void Xml_DefaultValueAttributeSetToNaNTest()
+    {
+        var xmlSerializer = new XmlSerializer(typeof(DefaultValuesSetToNaN));
+        var originalValue = new DefaultValuesSetToNaN();
+        using (var ms = new MemoryStream())
+        {
+            xmlSerializer.Serialize(ms, originalValue);
+            ms.Flush();
+            ms.Position = 0;
+            var roundtrippedValue = xmlSerializer.Deserialize(ms);
+            Assert.True(originalValue.Equals(roundtrippedValue), "originalValue doesn't equal the roundtrippedValue object!");
+        }
+    }
+
+    [Fact]
+    public static void Xml_NullRefInXmlSerializerCtorTest()
+    {
+        string defaultNamespace = "http://www.contoso.com";
+        var stream = new MemoryStream();
+        var order = PurchaseOrder.CreateInstance();
+        var serializer = new XmlSerializer(order.GetType(), null, null, null, defaultNamespace);
```

Comment:
To use `SerializeAndDeserialize()` for this test, you would need to provide a `Func` which return the serializer. The following probably is what you would need,

```c#
var actual = SerializeAndDeserialize(
    order, 
    <baseline string>, 
    () => new XmlSerializer(order.GetType(), null, null, null, defaultNamespace)
);
```